<a href="https://colab.research.google.com/github/sdasmaha/Ml_Engineers_445/blob/master/Diabetes_Notebook_Exercise1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

 Notebook Excercise #1 - SRINJOY DASMAHAPATRA

 IDE => Google Colab


#Importing Libraries

In [ ]:
import numpy as np
import pandas as pd

#Mounting the Google Drive


In [ ]:
from google.colab import drive
drive.mount("/content/gdrive")

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).



#Loading the Dataset


In [ ]:
diabetes_df = pd.read_csv('/content/gdrive/My Drive/Colab Notebooks/BAJWA-ML/diabetes_dataset.csv')

# *2.1*

**A**

In [ ]:
print(diabetes_df.shape) 

(768, 9)


In [ ]:
print(diabetes_df.axes)

[RangeIndex(start=0, stop=768, step=1), Index(['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Insulin',
       'BMI', 'DiabetesPedigreeFunction', 'Age', 'Outcome'],
      dtype='object')]


In [ ]:
print(diabetes_df.dtypes)

Pregnancies                   int64
Glucose                       int64
BloodPressure                 int64
SkinThickness                 int64
Insulin                       int64
BMI                         float64
DiabetesPedigreeFunction    float64
Age                           int64
Outcome                       int64
dtype: object


**B**

In [ ]:
diabetes_df.head(10)

Pregnancies  Glucose  BloodPressure  ...  DiabetesPedigreeFunction  Age  Outcome
0            6      148             72  ...                     0.627   50        1
1            1       85             66  ...                     0.351   31        0
2            8      183             64  ...                     0.672   32        1
3            1       89             66  ...                     0.167   21        0
4            0      137             40  ...                     2.288   33        1
5            5      116             74  ...                     0.201   30        0
6            3       78             50  ...                     0.248   26        1
7           10      115              0  ...                     0.134   29        0
8            2      197             70  ...                     0.158   53        1
9            8      125             96  ...                     0.232   54        1

[10 rows x 9 columns]

**C**    Based on the knowledge of the dataset and its stated usage, we are dealing with supervised problem. 

Since in a supervised learning model, the algorithm learns on a labeled dataset, providing an answer key that the algorithm can use to evaluate its accuaracy on training data. 

On the contrary the unsupervised model works on unlabeled data that the algorithm tries to make sense of by extracting features and patterns on its own.


**D** The independent variables in this case are  labels- 

1. Pregnancies	
2. Glucose
3. BloodPressure
4. SkinThickness
5. Insulin
6. BMI
7. DiabetesPedigreeFunction
8. Age




This is so because these variable/labels embedded in the algorithm 
increase or decrease the probability/likhehood of the event
"*Outcome*".


**E** The dependent variable in this case is the label *Outcome* since it is the conclusive/final event or variable that predicts the likelihood of the occurence of diabetes.

**F** Here the categorical variables are - *anaemia, diabetes,

1.   List item
2.   List item

high_blood_pressure, sex, smoking, DEATH_EVENT*.

This is so because a categorical variable is a variable that can take on one of a limited, and usually fixed, number of possible values, assigning each individual or other unit of observation to a particular group or nominal category on the basis of some qualitative property.



**G** The categorical variables follow *Label Encoding* where each label is mapped to a unique quantitative value.

**H**

In [ ]:
pd.cut(x=diabetes_df['Age'], bins=[21, 40, 60, 110]).value_counts()

(21, 40]     511
(40, 60]     167
(60, 110]     27
Name: Age, dtype: int64

Young Adults : 511 samples

1.   Young Adults : 511 samples
2.   Middle-ages Adults : 167 samples
3.   Old-aged Adults: 27

